### ISSUE TITLE
Page cache does not work under windows

### ISSUE THREAD

**Issue by fritzmg** [contao/core-bundle#464]
When trying to open a page in the frontend, for which the page cache is enabled and no cache file is present yet, i.e. when contao tries to create the page cache, the following errors appear:
```
Warning: mkdir(): Invalid argument in C:\xampp\htdocs\c41dev\vendor\contao\core-bundle\src\Resources\contao\library\Contao\Files.php on line 87

Warning: mkdir(): Invalid argument in C:\xampp\htdocs\c41dev\vendor\contao\core-bundle\src\Resources\contao\library\Contao\Files.php on line 87

Warning: mkdir(): Invalid argument in C:\xampp\htdocs\c41dev\vendor\contao\core-bundle\src\Resources\contao\library\Contao\Files.php on line 87

Warning: mkdir(): Invalid argument in C:\xampp\htdocs\c41dev\vendor\contao\core-bundle\src\Resources\contao\library\Contao\Files.php on line 87

Warning: rename(C:\xampp\htdocs\c41dev/system/tmp/8e36b279c44f8876dc283b8cc4a811d1,C:\xampp\htdocs\c41dev/C:\xampp\htdocs\c41dev\app\cache\prod/contao/html/6/624597f1b0ac07f979f574afbf5bd363.html): The filename, directory name, or volume label syntax is incorrect. (code: 123) in C:\xampp\htdocs\c41dev\vendor\contao\core-bundle\src\Resources\contao\library\Contao\Files.php on line 217

Warning: mkdir(): Invalid argument in C:\xampp\htdocs\c41dev\vendor\contao\core-bundle\src\Resources\contao\library\Contao\Files.php on line 87

Warning: mkdir(): Invalid argument in C:\xampp\htdocs\c41dev\vendor\contao\core-bundle\src\Resources\contao\library\Contao\Files.php on line 87

Warning: mkdir(): Invalid argument in C:\xampp\htdocs\c41dev\vendor\contao\core-bundle\src\Resources\contao\library\Contao\Files.php on line 87

Warning: mkdir(): Invalid argument in C:\xampp\htdocs\c41dev\vendor\contao\core-bundle\src\Resources\contao\library\Contao\Files.php on line 87

Warning: rename(C:\xampp\htdocs\c41dev/system/tmp/84284e9ce16842e877261c5479e52bc5,C:\xampp\htdocs\c41dev/C:\xampp\htdocs\c41dev\app\cache\prod/contao/html/6/624597f1b0ac07f979f574afbf5bd363.html): The filename, directory name, or volume label syntax is incorrect. (code: 123) in C:\xampp\htdocs\c41dev\vendor\contao\core-bundle\src\Resources\contao\library\Contao\Files.php on line 217
```
Again, it appears that Contao is trying to remove and add the `TL_ROOT` with the wrong directory separators, since the `rename` function call has this as one of its parameters:
```
C:\xampp\htdocs\c41dev/C:\xampp\htdocs\c41dev\app\cache\prod/contao/html/6/624597f1b0ac07f979f574afbf5bd363.html
```

**Comment by fritzmg** [comment]
This is may be related to #461. As pointed out in that issue
```php
$strCacheDir = str_replace(TL_ROOT . '/', '', \System::getContainer()->getParameter('kernel.cache_dir'));
```
Does not remove the root directory from the `kernel.cache_dir`. Thus, within the constructor of `Contao\System`, the folders cannot be created, since the constructor prepends `TL_ROOT` again.

**Comment by leofeyer** [comment]
Fixed in 5040b0fde75bdeddafedc486233294c411ab73cf.
